typings/styled-components.d.ts

```diff
@@ -199,6 +199,8 @@ export interface ThemedStyledComponentsModule<T> {
 declare const styled: StyledInterface;
 
 export const css: ThemedCssFunction<any>;
+export function withTheme<C extends Component<any>>(component: C): C;
```

Comment:
1. it does not reflect that the resulting component does not have `theme` prop
2. component parameter must have a `theme` prop
3. has no strong type theming support (`theme` prop must be of the same type as theme)
4. we do not need to capture component type and return it, we only need to capture props type and always return `ComponentClass`